feat: display team name in set_pipeline step header

analytically opened the pull request:
Add team field to BuildStepSetPipeline type and display it in the step header after the pipeline name when specified. Backend already sends the team field; frontend was ignoring it.

Fixes #9255

taylorsilva reviewed (changes requested):
When you're making UI changes, can you please include before/after images in your PR description. It helps speed up reviewing the PR.

Before
<img width="1960" height="1456" alt="image" src="https://github.com/user-attachments/assets/6e2573fd-ed3e-4975-ab14-93d6671b64fa" />

After
<img width="1960" height="1456" alt="image" src="https://github.com/user-attachments/assets/c738b47c-677e-4395-8fdc-09a59102bfc2" />

---

So after the pipeline name you've appended `[TEAM]`. It doesn't feel like this meshes with the style we usually see. For example, with the across step:

<img width="1934" height="692" alt="image" src="https://github.com/user-attachments/assets/a122c821-d957-4dcb-b312-c9301a2bf53a" />

Which visually feels easier to parse. Initially I couldn't even tell where the team names were rendering.

I'm thinking this:
<img width="628" height="88" alt="image" src="https://github.com/user-attachments/assets/3d93839b-de04-4faa-84e7-8bef572bbf16" />

Instead of what you currently have here:
<img width="542" height="90" alt="image" src="https://github.com/user-attachments/assets/95b7d886-aa08-47cd-94aa-2dcf54247074" />

I'm open to other visual styles if you've got any other ideas. WDYT?

analytically commented:
Okay I'll change the style to be more in line with other places. I'll post screenshots.

analytically committed:
feat: display team name in set_pipeline step header

Add team field to BuildStepSetPipeline type and display it in the step
header after the pipeline name when specified. Backend already sends
the team field; frontend was ignoring it.

Fixes #9255

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

taylorsilva reviewed (changes requested):
Yes, looks good now! I like that it only shows the `team` field if it's different from the current team.

<img width="750" height="708" alt="image" src="https://github.com/user-attachments/assets/153e3fb6-8135-41a3-8d37-adf76c4f5bce" />

Just one small thing to fix from `yarn analyse` and then this is good to merge.

analytically committed:
Update web/elm/src/Build/StepTree/StepTree.elm

Co-authored-by: Taylor Silva <dev@taydev.net>
